### PR TITLE
Improve error message for Failed Block Reads

### PR DESCRIPTION
**Pull request by apc999**
Confusing error message on failed reads are observed. This PR improves or corrects the following items:
- Throw the right exception on failed `openBlock(context, response);`, rather than always complaining missing fallback blocks
- Implement `toString` for `BlockReadRequest`, otherwise `request alluxio.worker.grpc.BlockReadRequest@1becc9d9` is not very helpful.
- Add Job Id in Job master logs so we can correlate Job master logs and master logs much easier  by the job Id

Before

```
On Worker:
2019-10-29 14:48:40,984 DEBUG AbstractReadHandler - Exception occurred while reading data for read request alluxio.worker.grpc.BlockReadRequest@1becc9d9.
java.io.FileNotFoundException: /Users/binfan/Dropbox/Work/alluxio/alluxio/underFSStorage/.alluxio_ufs_blocks.alluxio.0x1D91AC0E01AB0165.tmp/1325400064 (No such file or directory)

On Master:
2019-10-29 14:48:42,718 WARN  DefaultFileSystemMaster - The persist job (id=1572385668302) for file /cache/part-00083 (id=1426063359) failed: Task execution failed: /Users/binfan/Dropbox/Work/alluxio/alluxio/underFSStorage/.alluxio_ufs_blocks.alluxio.0x1D91AC0E01AB0165.tmp/1409286144 (No such file or directory) (GrpcDataReader{request=block_id: 1409286144

on Job Master
2019-10-29 17:11:49,804 INFO  job.JobCoordinator (JobCoordinator.java:start) - Starting job PersistConfig{filePath=/cache/part-00053, mountId=1, overwrite=false, ufsPath=/Users/binfan/Dropbox/Work/alluxio/alluxio/underFSStorage/cache/part-00053.alluxio.0x0000016E1A0118AA.tmp}
```

After

```
On Worker:
2019-10-29 17:11:50,517 WARN  AbstractReadHandler - Exception occurred while reading data for read request BlockReadRequest{chunkSize=1048576, end=33554479, id=939524096, openUfsBlockOptions=offset_in_file: 0
block_size: 33554479
maxUfsReadConcurrency: 2147483647
mountId: 1
no_cache: true
block_in_ufs_tier: true
, promote=false, sessionId=1117586808638664750, start=0}.: Failed to open block id=939524096

On Master
2019-10-29 17:11:51,771 WARN  DefaultFileSystemMaster - The persist job (id=1572389450770) for file /cache/part-00056 (id=973078527) failed: Task execution failed: Failed to open block id=956301312 (GrpcDataReader{request=block_id: 956301312

On Job Master
2019-10-29 17:11:49,804 INFO  job.JobCoordinator (JobCoordinator.java:start) - Starting job Id=1572389450769 Config=PersistConfig{filePath=/cache/part-00053, mountId=1, overwrite=false, ufsPath=/Users/binfan/Dropbox/Work/alluxio/alluxio/underFSStorage/cache/part-00053.alluxio.0x0000016E1A0118AA.tmp}
```

### CODE REVIEW

**Comment by gpang** [comment]
`/usr/src/alluxio/core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java:175:41: error: '&&' should be on a new line.`

File: core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java [hunk 1]
```diff
@@ -167,7 +172,8 @@ private void openBlock(BlockReadRequestContext context, StreamObserver<ReadRespo
 
         // When the block does not exist in Alluxio but exists in UFS, try to open the UFS block.
         Protocol.OpenUfsBlockOptions openUfsBlockOptions = request.getOpenUfsBlockOptions();
-        if (mWorker.openUfsBlock(request.getSessionId(), request.getId(),
+        if (openUfsBlockOptions != null &&
```
Comment (calvinjia) [inline core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java]: Protobuf does not allow null values so this check is unnecessary?

Comment (apc999) [reply core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java]: done

**Comment by apc999** [comment]
@calvinjia PTAL

**Comment by apc999** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
@calvinjia I took the approach to be more explicit in the error message on both failed tries to minimize the footprint. PTAL

File: core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java [hunk 2]
```diff
@@ -183,14 +183,16 @@ private void openBlock(BlockReadRequestContext context, StreamObserver<ReadRespo
...
-            throw e;
           }
+        } catch (Exception e) {
+          // TODO(binfan): remove the closeUfsBlock here as the exception will be handled in
+          // AbstractReadHandler. Current approach to use context.blockReader as a flag is a
+          // workaround.
+          mWorker.closeUfsBlock(request.getSessionId(), request.getId());
+          context.setBlockReader(null);
+          throw new UnavailableException(String.format("Failed to read block id=%s at "
+                  + "neither tiered storage nor potential UFS tier: %s",
```
Comment (calvinjia) [inline core/server/worker/src/main/java/alluxio/worker/grpc/BlockReadHandler.java]: `Failed to read block id=%s from tiered storage and UFS tier: %s`

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Pull request merged**